src/sql-parser/src/parser.rs

```diff
@@ -2900,7 +2900,22 @@ impl<'a> Parser<'a> {
     fn parse_create_sink(&mut self) -> Result<Statement<Raw>, ParserError> {
         self.expect_keyword(SINK)?;
         let if_not_exists = self.parse_if_not_exists()?;
-        let name = self.parse_item_name()?;
+
+        let mut name = Some(self.parse_item_name()?);
+
+        // Sniff out `CREATE SINK IN CLUSTER <c> ...` and `CREATE SINK FROM
+        // <view>...`  and ensure they are parsed as nameless `CREATE SINK`
+        // commands.
+        //
+        // This is a bit gross, but we didn't have the foresight to make
+        // `IN` and `FROM` reserved keywords for sink names.
```

Comment:
Even if we made `IN` and `FROM` keywords, it would still be an issue because `parse_item_name` eventually is a call to `consume_identifier` where we do accept keywords as a valid identifier.

https://github.com/MaterializeInc/materialize/blob/v0.76.0/src/sql-parser/src/parser.rs#L5468-L5478

```
fn consume_identifier(&mut self) -> Option<Ident> {
        match self.peek_token() {
            Some(Token::Keyword(kw)) => {
                self.next_token();
                Some(kw.into())
            }
            Some(Token::Ident(id)) => {
                self.next_token();
                Some(Ident::new(id))
            }
            _ => None,
        }
    }
```

Reply:
Right, `parse_item_name` would need to learn to respect reserved-ness like `parse_optional_alias` does.